docs/cookbook/using-docker.md

```diff
@@ -0,0 +1,321 @@
+
+<AposCodeBlock>
+
+```bash
+.vscode/
+apos-build/
+badges/
+data/
+node-modules/
+public/uploads/
+.dockerignore
+.env
+.eslintignore
+.gitignore
+deploy-test-count
+docker-compose.yaml
+dockerfile
+force-deploy
+local.example.js
+```
+
+<template v-slot:caption>
+  .dockerignore
+</template>
+
+</AposCodeBlock>
+
+### Creating a `docker-compose.yaml` file
+In this guide, we are starting by creating multiple containers and a persistent volume. This is so that we can provide both a MongoDB instance and a place to store uploaded assets. We are going to do this using [Docker Compose](https://docs.docker.com/compose/) and a `docker-compose.yml` file. In the following sections of the tutorial, we will look at removing the extra container and volume by taking advantage of cloud storage and database services. Create this file at the root of your project.
+
+<AposCodeBlock>
+
+```bash
+services:
+  db:
+    image: mongo:4.4.14
+    ports:
+      - "27018:27018"
+    volumes:
+      - /data/db
+  web:
+    build:
+      context: .
+    container_name: "apostrophe-container"
+    ports:
+      - "3000:3000"
+    environment:
+      - NODE_ENV
+      - APOS_MONGODB_URI
+      - APOS_CLUSTER_PROCESSES
+    depends_on:
+      - db 
+    volumes:
+      - /srv/www/apostrophe/public/uploads
+```
+
+<template v-slot:caption>
+  docker-compose.yaml
+</template>
+
+</AposCodeBlock>
+
+The spacing in this file is very important. Whitespace, not tab, indentation indicates that a particular line is nested within the object passed on the line above it. Walking through this file, it starts with `services:`. From the indentation, we can see that we are creating two services - a `db:` container and a `web:` container.  Much like our `Dockerfile`, within the `db:` we start by specifying an image to run. In this case, it is the `mongo:4.4.14` official image for running MongoDB v4.4.14. Other images can be found in the docker library GitHub repo [README](https://github.com/docker-library/docs/blob/master/mongo/README.md#supported-tags-and-respective-dockerfile-links). You should use the version that mirrors your development environment.
+
+Next, we are specifying that the database should communicate over port `27018`. This is different from the port typically used in order to direct communication to the dockerized version and not a local MongoDb. If you need your database to communicate over a different port, you have to change it here and in your `.env` file.
+
+Finally, we add a volume for the MongoDB storage engine to write files into. You shouldn't need to change this.
```

Comment:
Without the persistent volume, the database would appear to work, but all content would be lost on every restart.

Reply:
Added.